src/main/java/com/bieganski/ox/Coordinates.java

```diff
@@ -0,0 +1,42 @@
+package com.bieganski.ox;
+
+import java.util.Objects;
+
+public class Coordinates implements Comparable<Coordinates> {
+    private int x;
+    private int y;
+
+    public Coordinates(int x, int y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Coordinates that = (Coordinates) o;
+        System.out.println(that);
+        return x == that.x &&
+                y == that.y;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(x, y);
+    }
+
+    @Override
+    public int compareTo(Coordinates coordinates) {
+        int compareY = Integer.compare(coordinates.y, y);
+        return compareY == 0 ? Integer.compare(coordinates.x, x) : compareY;
+    }
+
```

Comment:
I remember there is some method Comparator.comparing().thenComparing() or something. I do not remember exactly how it works but if you are interested you can check if it can be applied here